### PR TITLE
Support outputting result in Json format

### PR DESCRIPTION
**Pull request by TheKK**
This should solve #23.

### CODE REVIEW

File: mozbench/mozbench.py [hunk 1]
```diff
@@ -36,6 +36,105 @@
 headers = None
 results = None
 
+class ResultRecorder(object):
```
Comment (dminor) [inline mozbench/mozbench.py]: Please move this class to it's own file.

File: mozbench/mozbench.py [hunk 2]
```diff
@@ -36,6 +36,105 @@
...
+    def __init__(self):
+        self._platform = 'unknown'
+        self._osVersion = 'unknown'
+        self._processor = 'unknown'
+        self._currentBrowser = None
+        self._currentBenchmark = None
+        self._browsers = {}
+
+    def setPlatformName(self, name):
+        self._platform = name
```
Comment (dminor) [inline mozbench/mozbench.py]: The python style is to not write simple getter/setter methods like this. Please just have a self.platform member and remove these functions.

File: mozbench/mozbench.py [hunk 3]
```diff
@@ -36,6 +36,105 @@
...
+
+    def setOSVersion(self, version):
+        self._osVersion = version
+
+    def setProcessorName(self, name):
+        self._processor = name
+
+    def setBrowser(self, browser):
```
Comment (dminor) [inline mozbench/mozbench.py]: Python uses underscores for method names, so this should be set_browser.

File: mozbench/mozbench.py [hunk 4]
```diff
@@ -36,6 +36,105 @@
...
+        self._osVersion = 'unknown'
+        self._processor = 'unknown'
+        self._currentBrowser = None
+        self._currentBenchmark = None
+        self._browsers = {}
+
+    def setPlatformName(self, name):
+        self._platform = name
+
+    def setOSVersion(self, version):
```
Comment (dminor) [inline mozbench/mozbench.py]: Please remove

File: mozbench/mozbench.py [hunk 5]
```diff
@@ -36,6 +36,105 @@
...
+        self._osVersion = version
+
+    def setProcessorName(self, name):
```
Comment (dminor) [inline mozbench/mozbench.py]: Please remove

File: mozbench/mozbench.py [hunk 6]
```diff
@@ -36,6 +36,105 @@
...
+
+    def setBrowser(self, browser):
+        if self._browsers.get(browser) is None:
+            self._browsers[browser] = {}
+            self._browsers[browser]['benchmarks'] = {}
+            self._browsers[browser]['version'] = ''
+
+        self._currentBrowser = self._browsers[browser]
+
+    def setBrowserVersion(self, version):
```
Comment (dminor) [inline mozbench/mozbench.py]: As above, set_browser_version

File: mozbench/mozbench.py [hunk 7]
```diff
@@ -36,6 +36,105 @@
...
+        if self._currentBrowser is None: raise AssertionError('You should setBrowser first')
+
+        self._currentBrowser['version'] = version or 'unknown'
+
+    def setBenchmark(self, benchmark):
```
Comment (dminor) [inline mozbench/mozbench.py]: set_benchmark

File: mozbench/mozbench.py [hunk 8]
```diff
@@ -36,6 +36,105 @@
 headers = None
 results = None
 
+class ResultRecorder(object):
+
+    def __init__(self):
+        self._platform = 'unknown'
+        self._osVersion = 'unknown'
+        self._processor = 'unknown'
+        self._currentBrowser = None
```
Comment (dminor) [inline mozbench/mozbench.py]: Class members should have underscores in their names, so this should be current_browser.

File: mozbench/mozbench.py [hunk 9]
```diff
@@ -36,6 +36,105 @@
...
+        self._currentBenchmark = None
```
Comment (dminor) [inline mozbench/mozbench.py]: current_benchmark

File: mozbench/mozbench.py [hunk 10]
```diff
@@ -36,6 +36,105 @@
...
+
+        if self._currentBrowser['benchmarks'].get(benchmark) is None:
+            self._currentBrowser['benchmarks'][benchmark] = {}
+            self._currentBrowser['benchmarks'][benchmark]['resultName'] = ''
+            self._currentBrowser['benchmarks'][benchmark]['resultValueName'] = ''
+            self._currentBrowser['benchmarks'][benchmark]['results'] = []
+
+        self._currentBenchmark = self._currentBrowser['benchmarks'][benchmark]
+
+    def setResultName(self, name):
```
Comment (dminor) [inline mozbench/mozbench.py]: set_result_name

File: mozbench/mozbench.py [hunk 11]
```diff
@@ -36,6 +36,105 @@
...
+        if self._currentBenchmark is None: raise AssertionError('You should setBenchmark first')
+
+        self._currentBenchmark['resultName'] = name
+
+    def setResultValueName(self, name):
```
Comment (dminor) [inline mozbench/mozbench.py]: set_result_valuename

File: mozbench/mozbench.py [hunk 12]
```diff
@@ -36,6 +36,105 @@
...
+        if self._currentBenchmark is None: raise AssertionError('You should setBenchmark first')
+
+        self._currentBenchmark['resultValueName'] = name
+
+    def addResults(self, results):
```
Comment (dminor) [inline mozbench/mozbench.py]: add_results

File: mozbench/mozbench.py [hunk 13]
```diff
@@ -36,6 +36,105 @@
...
+        if self._currentBenchmark is None: raise AssertionError('You should setBenchmark first')
+
+        self._currentBenchmark['results'].append(copy.copy(results))
+
+    def getInfluxDBResults(self):
```
Comment (dminor) [inline mozbench/mozbench.py]: get_influxdb_results

File: mozbench/mozbench.py [hunk 14]
```diff
@@ -36,6 +36,105 @@
...
+                        resultPoint = {
+                                'name': table,
+                                'columns': ['value','browser-version', 'os-version', 'processor'],
+                                'points': [[value, browserVersion, osVersion, processor]]
+                        }
+                        resultsToReturn.append(resultPoint)
+
+        return resultsToReturn
+
+    def getGeneralResults(self):
```
Comment (dminor) [inline mozbench/mozbench.py]: I think get_results would be fine for this

**Comment by dminor** [comment]
Thanks, this looks good overall, just some small coding style changes and I'll be happy to merge this in.

**Comment by TheKK** [comment]
Now codes should in appropriate style, thanks for your comments and feedbacks :)

**Comment by dminor** [comment]
Thanks for making those changes. Squashed and merged as 509020ccec5ba579dfcab5998227a8e2654e251b

**Pull request closed without merging**